src/employee-invoice/service/employee-invoice/employee-salary.service.ts

```diff
@@ -0,0 +1,71 @@
+import { HttpException, HttpStatus, Injectable, NotFoundException } from '@nestjs/common';
+import { Prisma } from '@prisma/client';
+import { EmployeeSalaryDto } from '../../dtos/employee-salary.dto';
+import { PrismaService } from '../../../prisma/prisma.service';
+
+@Injectable()
+export class EmployeeSalaryService {
+    constructor(private readonly prisma: PrismaService) { }
+
+    async createInvoice(data: Prisma.EmployeeSalarySlipCreateManyInput) {
+        const totalAmount = data.basicSalary + data.insentive +
+            (data.overTime * ((data.basicSalary / 30) / 9)) -
```

Comment:
Store this in seperate varible and reuse all place
data.basicSalary / 30) / 9

like

const hourlyRate = data.basicSalary / 30) / 9;

Note: Calculation based on selected month wise For example we are calculating for August-2023 so after remove all sat-sun we have 23 working days so hourly rate calculated on that

Reply:
done